tests/index.ts

```diff
@@ -0,0 +1,165 @@
+// - KnownGoodVersion, which parses the linter definition and attempts to specify a known_good_version
+// - Latest, which automatically retrieves the latest linter version if network connectivity is available
+// - A specified version. Note that this will apply to all tests, so only use this environment variable when tests are filtered
+const parseLinterVersion = (value: string): ILinterVersion | string | undefined => {
+  if (value == "KnownGoodVersion") {
+    return ILinterVersion.KnownGoodVersion;
+  } else if (value == "Latest") {
+    return ILinterVersion.Latest;
+  } else if (value && value.length > 0) {
+    return value;
+  }
+  return undefined;
+};
+
+// Parse the global testing config inputs, specified as environment variables.
+// - PLUGINS_TEST_CLI_VERSION replaces the repo-wide trunk.yaml's specified cli-version
+// - PLUGINS_TEST_CLI_PATH specifies an alternative path to a trunk binary
+// - PLUGINS_TEST_LINTER_VERSION specifies a linter version semantic (see 'parseLinterVersion')
+export const parseInputs = () => {
+  const args = <ITestingArguments>{
+    cliVersion: process.env.PLUGINS_TEST_CLI_VERSION,
+    cliPath: process.env.PLUGINS_TEST_CLI_PATH,
+    linterVersion: parseLinterVersion(process.env.PLUGINS_TEST_LINTER_VERSION ?? ""),
+  };
+  if (!printed_args && (args.cliVersion || args.cliPath || args.linterVersion)) {
+    console.debug(args);
+    printed_args = true;
+  }
+  return args;
+};
+
+// If 'namedTestPrefixes' are specified, checks for their existence in 'dirname'. Otherwise,
+// automatically scan 'dirname' for all available test input/output pairs.
+const detectTestTargets = (dirname: string, namedTestPrefixes: string[]): ITestTarget[] => {
+  const parentTestDirName = path.parse(dirname).name;
+  const testTargets = new Map<string, ITestTarget>();
+  // Sort guarantees basic.in comes before basic.out
+  fs.readdirSync(dirname)
+    .sort()
+    .forEach((file: string) => {
+      const inFileRegex = /(?<prefix>.+)\.in\.(?<extension>.+)$/;
+      const foundIn = file.match(inFileRegex);
+      if (foundIn) {
+        const prefix = foundIn.groups?.prefix;
+        if (prefix && (namedTestPrefixes.includes(prefix) || namedTestPrefixes.length == 0)) {
+          testTargets.set(prefix, {
+            prefix,
+            inputPath: path.join(parentTestDirName, file),
+            outputPath: "",
+          });
+        }
+        return;
+      }
+
+      const outFileRegex = /(?<prefix>.+)\.out\.(?<extension>.+)$/;
+      const foundOut = file.match(outFileRegex);
+      if (foundOut) {
+        const prefix = foundOut.groups?.prefix;
+        if (prefix) {
+          const maybe_target = testTargets.get(prefix);
+          if (maybe_target) {
+            maybe_target.outputPath = path.join(parentTestDirName, file);
+          }
+        }
+      }
+    });
+  return Array.from(testTargets.values()).filter((target) => target.outputPath.length > 0);
```

Comment:
nit: `Array.from` is fine, but spreading into an array is more terse
```suggestion
  return [...testTargets.values()].filter((target) => target.outputPath.length > 0);
```